modules/synthetic-protocol/src/lib.rs

```diff
@@ -242,11 +236,10 @@ impl<T: Trait> Module<T> {
 		let (collateral_position_delta, pool_refund_collateral) =
 			Self::_calc_remove_position(pool_id, currency_id, price, synthetic, redeemed_collateral)?;
 
-		ensure!(
-			T::CollateralCurrency::balance(&<SyntheticTokens<T>>::account_id())
-				>= redeemed_collateral + pool_refund_collateral,
-			Error::<T>::NotEnoughLockedCollateralAvailable,
-		);
+		T::CollateralCurrency::ensure_can_withdraw(
+			&<SyntheticTokens<T>>::account_id(),
+			redeemed_collateral + pool_refund_collateral,
+		)?;
```

Comment:
Map the error to `Error::<T>::NotEnoughLockedCollateralAvailable`. The fact that collateral is locked in synthetic tokens module account should not be exposed to user.